tests/end_to_end/gcc-division/orig.c

```diff
@@ -0,0 +1,64 @@
+void func_00400090(unsigned int x) {
+}
+
+void test(volatile int x, volatile unsigned int y) {
+    func_00400090(x / 2);
+    func_00400090(x / 3);
+    func_00400090(x / 4);
+    func_00400090(x / 5);
+    func_00400090(x / 6);
+    func_00400090(x / 7);
+    func_00400090(x / 8);
+    func_00400090(x / 9);
+    func_00400090(x / 10);
+    func_00400090(x / 100);
+    func_00400090(x / 255);
+    func_00400090(x / 256);
+    func_00400090(x / 360);
+    func_00400090(x / 1000);
+
+    func_00400090(x % 2);
+    func_00400090(x % 3);
+    func_00400090(x % 4);
+    func_00400090(x % 5);
+    func_00400090(x % 6);
+    func_00400090(x % 7);
+    func_00400090(x % 8);
+    func_00400090(x % 9);
+    func_00400090(x % 10);
+    func_00400090(x % 100);
+    func_00400090(x % 255);
+    func_00400090(x % 256);
+    func_00400090(x % 360);
+    func_00400090(x % 1000);
+
+    func_00400090(y / 2);
+    func_00400090(y / 3);
+    func_00400090(y / 4);
+    func_00400090(y / 5);
+    func_00400090(y / 6);
+    func_00400090(y / 7);
+    func_00400090(y / 8);
+    func_00400090(y / 9);
+    func_00400090(y / 10);
+    func_00400090(y / 100);
+    func_00400090(y / 255);
+    func_00400090(y / 256);
+    func_00400090(y / 360);
+    func_00400090(y / 1000);
+
+    func_00400090(y % 2);
+    func_00400090(y % 3);
+    func_00400090(y % 4);
+    func_00400090(y % 5);
+    func_00400090(y % 6);
+    func_00400090(y % 7);
+    func_00400090(y % 8);
+    func_00400090(y % 9);
+    func_00400090(y % 10);
+    func_00400090(y % 100);
+    func_00400090(y % 255);
+    func_00400090(y % 256);
+    func_00400090(y % 360);
+    func_00400090(y % 1000);
+}
```

Comment:
I wouldn't mind seeing this extended even further, at least for say signed int division. Maybe 11..33, 257, numbers closer to int max. And s8/s16 to test that branch.

Reply:
Added more test cases. I had to split them into separate functions, because GCC started putting the magic constants on the stack.